### PR TITLE
feat: BNGP-3912 Change Applicant info from CYA

### PR DESCRIPTION
**Pull request by KiranVarma02**

### CODE REVIEW

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]
In addition to my comments on individual files:

- If the user confirms they want to change applicant information it is possible for them to go back to the dashboard, work on another application before the replacement applicant information is complete. When the user returns to the application with partially modified applicant information, null and undefined data is shown on the check your answers page. For example:
image
- I don't think changing if applicant is an individual or organisation is compatible with https://github.com/DEFRA/biodiversity-net-gain-service/pull/529. Since the application dashboard is now filtered according to how the user logged in I don't believe it is possible to change whether the user is representing themselves or an organisation on a particular application. This provides rationale for reconsidering this area of functionality.

File: packages/webapp/src/routes/__tests__/land/change-client-individual-organisation.spec.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      const request = {
+        yar: redisMap,
+        payload: { changeClientIndividualOrganisation: 'yes' }
+      }
+
+      await changeClientIndividualOrganisation.default[1].handler(request, h)
+
+      expect(viewResult).toEqual(constants.routes.CLIENT_INDIVIDUAL_ORGANISATION)
+    })
+    it('Should continue journey to CHECK_APPLICANT_INFORMATION if user does not want to change applying as individual or organisation', async () => {
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/change-client-individual-organisation.spec.js:44]: What is wrong here?

File: packages/webapp/src/routes/land/change-applying-individual-organisation.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const handlers = {
+  get: async (_request, h) => {
+    return h.view(constants.views.CHANGE_APPLYING_INDIVIDUAL_ORGANISATION)
+  },
+  post: async (request, h) => {
+    const { changeApplyingIndividualOrganisation } = request.payload
+
+    if (changeApplyingIndividualOrganisation === 'yes') {
+      request.yar.clear(constants.redisKeys.LANDOWNER_TYPE)
+      request.yar.clear(constants.redisKeys.CLIENT_INDIVIDUAL_ORGANISATION_KEY)
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply packages/webapp/src/routes/land/change-applying-individual-organisation.js:12]: IS_AGENT is to clear behalf of a client, CLIENT_INDIVIDUAL_ORGANISATION_KEY is to clear individual or org. Seems okay to me. Please check again.

File: packages/webapp/src/routes/land/change-applying-individual-organisation.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      request.yar.clear(constants.redisKeys.IS_ADDRESS_UK_KEY)
+      request.yar.clear(constants.redisKeys.UK_ADDRESS_KEY)
+      request.yar.clear(constants.redisKeys.CLIENTS_NAME_KEY)
+      request.yar.clear(constants.redisKeys.CLIENTS_ORGANISATION_NAME_KEY)
+      request.yar.clear(constants.redisKeys.CLIENTS_EMAIL_ADDRESS_KEY)
+      request.yar.clear(constants.redisKeys.CLIENTS_PHONE_NUMBER_KEY)
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply packages/webapp/src/routes/land/change-applying-individual-organisation.js:18]: is this a duplicate comment? please check again.

File: packages/webapp/src/routes/__tests__/land/change-client-individual-organisation.spec.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  })
+
+  describe('GET', () => {
+    it(`should render the ${url.substring(1)} view`, async () => {
+      await submitGetRequest({ url })
+    })
+  })
+
+  describe('POST', () => {
+    it('Should continue journey to CLIENT_INDIVIDUAL_ORGANISATION if user confirms to changing legal agreement', async () => {
```
[the remark this answers is not in the document]
Comment (pwadmore-ea) [reply packages/webapp/src/routes/__tests__/land/change-client-individual-organisation.spec.js]: This pull request has nothing to do with legal agreements so the text in bold needs correcting.

File: packages/webapp/src/routes/__tests__/land/change-client-individual-organisation.spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (pwadmore-ea) [reply packages/webapp/src/routes/__tests__/land/change-client-individual-organisation.spec.js:44]: As the tests file in this file appear to be for an agent acting on behalf of an individual or organisation I suggested adding the text in bold.

File: packages/webapp/src/routes/land/change-applying-individual-organisation.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (pwadmore-ea) [reply packages/webapp/src/routes/land/change-applying-individual-organisation.js:12]: Thanks. I understand why the CLIENT_ keys are being cleared now.

File: packages/webapp/src/routes/land/change-applying-individual-organisation.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (pwadmore-ea) [reply packages/webapp/src/routes/land/change-applying-individual-organisation.js:18]: This comment is linked to the highlighted keys but I understand why they are being cleared now.

**Review by pwadmore-ea — CHANGES_REQUESTED** [review]
@KiranVarma02  - I've resolved some review comments following your comments and hope I've clarified the others.

File: packages/webapp/src/routes/__tests__/land/change-client-individual-organisation.spec.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/change-client-individual-organisation.spec.js:44]: done

File: packages/webapp/src/routes/__tests__/land/change-applying-individual-organisation.spec.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  })
+
+  describe('GET', () => {
+    it(`should render the ${url.substring(1)} view`, async () => {
+      await submitGetRequest({ url })
+    })
+  })
+
+  describe('POST', () => {
+    it('Should continue journey to APPLICATION_BY_INDIVIDUAL_OR_ORGANISATION if user confirms to changing legal agreement', async () => {
```
[the remark this answers is not in the document]
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/change-applying-individual-organisation.spec.js:34]: done

[the remark this answers is not in the document]
Comment (pwadmore-ea) [reply packages/webapp/src/routes/__tests__/land/change-applying-individual-organisation.spec.js]: @KiranVarma02  - **legal agreement** is the typo that needs changing as this is not a legal agreement story.

[the remark this answers is not in the document]
Comment (KiranVarma02) [reply packages/webapp/src/routes/__tests__/land/change-applying-individual-organisation.spec.js]: this says outdated, it was corrected when you put first comment. Do you not see update? not sure whats happening!

[the remark this answers is not in the document]
Comment (pwadmore-ea) [reply packages/webapp/src/routes/__tests__/land/change-applying-individual-organisation.spec.js]: As discussed I commented on the wrong file earlier and will resolve this conversation.

**Review by pwadmore-ea — APPROVED** [review]

**Comment by shaundefra** [comment]
tested by @LeenaKu

**Pull request merged**